libs/libskynet/.eslintrc.json

```diff
@@ -1,18 +1,18 @@
 {
-	"env": {
-		"browser": true,
-		"es2021": true,
-		"node": true
-	},
-	"extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
-	"parser": "@typescript-eslint/parser",
-	"parserOptions": {
-		"ecmaVersion": "latest",
-		"sourceType": "module"
-	},
-	"plugins": ["@typescript-eslint"],
-	"rules": {
-		"@typescript-eslint/no-explicit-any": "off",
-		"@typescript-eslint/no-non-null-assertion": "off"
-	}
+  "env": {
+    "browser": true,
+    "es2021": true,
+    "node": true
+  },
+  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
+  "parser": "@typescript-eslint/parser",
+  "parserOptions": {
+    "ecmaVersion": "latest",
+    "sourceType": "module"
+  },
+  "plugins": ["@typescript-eslint"],
+  "rules": {
+    "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/no-non-null-assertion": "off"
```

Comment:
Why disable that? It prevents one of the most common bugs from happening (`cannot read property x of undefined`).

Reply:
There are a ton of places in the code where we deal with untrusted objects, and typescript is not very effective at determining whether or not we've already checked that an object is okay. This is made worse by the fact that we deal with objects like `MessageEvents`, which have optional fields that we know will always be available in the contexts that we use them, but typescript doesn't understand that we are in a safe context.

If we let `no-non-null-assertion` run, we end up needing to do a bunch of frivolous type checking even though we already know the object is safe.

Reply:
In general I'm in favor of a check like this, I just think typescript is too stupid for this check to be worthwhile.